debezium-core/src/main/java/io/debezium/transforms/ExtractNewRecordState.java

```diff
@@ -312,29 +313,27 @@ public void close() {
          * The prefix for the new field name.
          */
         private final String prefix;
+
         /**
          * The name for the outgoing attribute/field, e.g. "__op" or "__source_ts_ms" when the prefix is "__"
          */
-        private final String newFieldName;
+        private final String newField;
 
         private FieldReference(String prefix, String field) {
             this.prefix = prefix;
             String[] parts = FIELD_SEPARATOR.split(field);
 
             if (parts.length == 1) {
-                this.struct = determineStruct(parts[0]);
-                this.field = parts[0];
-                this.newFieldName = prefix + field;
+                String[] splits = NEW_FIELD_SEPARATOR.split(parts[0]);
```

Comment:
@rgannu Hi, I think it is mostly correctly. I am certain doubts about this part of the code. While it works I'd prefer to do it more logically. IMHO first `NEW_FIELD_SEPARATOR` should be applied as it defines the main components to which the mapping splits. And only after that `FIELD_SEPARATOR` should be applied.

The reason is that the new field name applies to fully qualified field name in the first part not only to one of the compoenents.

Reply:
Agreed and I will fix this in a separate branch and create a PR. 

Also I will create a new ticket that the prefix configuration for header and property needs to be a `list` instead of a `string`. Then, we can apply different prefixes for different field when needed. When it is not defined the default prefix of double-underscore will continue.

Reply:
@rgannu May I ask if you know a timeframe for the last change? Thanks

Reply:
Sorry, i was quite busy with my office work.. Will do for sure by eod today...

Reply:
No worries, take your time

Reply:
Fixed it .. Thanks @jpechane  .. 
The code looks much clean..

Also raised a new enhancement DBZ-2695 for making the "add.fields.prefix" and "add.headers.prefix" as a list of strings instead of just a string.